src/control/server/mgmt_system.go

```diff
@@ -289,6 +305,17 @@ func (svc *mgmtSvc) join(ctx context.Context, req *batchJoinRequest) *batchJoinR
 	return resp
 }
 
+// reqGroupUpdate requests an asynchronous group update.
+func (svc *mgmtSvc) reqGroupUpdate(ctx context.Context) {
+	select {
```

Comment:
do you really need to select on context done when sending to an unbuffered channel? isn't that a nonblocking operation that will return immediately

Reply:
@mjmac explained that this is to cover the case where the receiver goroutine is killed before the sender tries to send to the channel, resulting in a block on channel write

Reply:
Right. To elaborate a bit further, sending to an unbuffered channel is only nonblocking if there is a receiver that is currently waiting on the channel. In this case, we want to avoid races on shutdown where the receiving goroutine is terminated before the sending goroutine is terminated, as this will cause the sender to block forever.